packages/flutter/lib/src/material/menu_bar_theme.dart

```diff
@@ -0,0 +1,333 @@
+/// overall [Theme] with [ThemeData.menuBarTheme]. Otherwise,
+/// [MenuBarTheme] can be used to configure its own widget subtree.
+///
+/// All [MenuBarThemeData] properties are `null` by default.
+/// If any of these properties are null, the menu bar will provide its own
+/// defaults.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+@immutable
+class MenuBarThemeData with Diagnosticable {
+  /// Creates the set of properties used to configure [MenuBarTheme].
+  const MenuBarThemeData({
+    this.barHeight,
+    this.barPadding,
+    this.barBackgroundColor,
+    this.barElevation,
+    this.menuBackgroundColor,
+    this.menuElevation,
+    this.menuShape,
+    this.menuPadding,
+    this.itemBackgroundColor,
+    this.itemForegroundColor,
+    this.itemOverlayColor,
+    this.itemTextStyle,
+    this.itemPadding,
+    this.itemShape,
+  });
+
+  /// The minimum height of the menu bar.
+  final double? barHeight;
+
+  /// The padding around the outside of a [MenuBar].
+  final EdgeInsets? barPadding;
+
+  /// The background color of the [MenuBar].
+  final MaterialStateProperty<Color?>? barBackgroundColor;
+
+  /// The Material elevation of the [MenuBar].
+  ///
+  /// See also:
+  ///
+  ///  * [Material.elevation] for a description of how elevation works.
+  final MaterialStateProperty<double?>? barElevation;
+
+  /// The background color of a [MenuBarMenu].
+  final MaterialStateProperty<Color?>? menuBackgroundColor;
+
+  /// The Material elevation of the [MenuBarMenu].
+  ///
+  /// See also:
+  ///
+  ///  * [Material.elevation] for a description of how elevation works.
+  final MaterialStateProperty<double?>? menuElevation;
+
+  /// The shape around a [MenuBarMenu].
+  final MaterialStateProperty<ShapeBorder?>? menuShape;
+
+  /// The padding around the outside of a [MenuBarMenu].
+  final EdgeInsets? menuPadding;
+
+  // MenuBarItem properties
+
+  /// The background color of a [MenuBarButton].
+  final MaterialStateProperty<Color?>? itemBackgroundColor;
+
+  /// The foreground color of a [MenuBarButton], used to color the text and
+  /// shortcut labels of a [MenuBarButton].
+  final MaterialStateProperty<Color?>? itemForegroundColor;
+
+  /// The overlay color of a [MenuBarButton], used to color the overlay of a
+  /// [MenuBarButton], typically seen when the [MaterialState.hovered],
+  /// [MaterialState.selected], and/or [MaterialState.focused] states apply.
+  final MaterialStateProperty<Color?>? itemOverlayColor;
+
+  /// The text style of the [MenuBarButton]s in a [MenuBar].
+  ///
+  /// The color in this text style will only be used if [itemForegroundColor]
+  /// is unset.
+  final MaterialStateProperty<TextStyle?>? itemTextStyle;
+
+  /// The padding around the outside of an individual [MenuBarButton].
+  final EdgeInsets? itemPadding;
+
+  /// The shape around an individual [MenuBarButton].
```

Comment:
The shape of an individual MenuBarButton.

Do we really want to support items whose shape changes based on their state?